app/code/ZP/LoyaltyProgram/Controller/Adminhtml/Program/Save.php

```diff
@@ -0,0 +1,171 @@
+<?php
+declare(strict_types=1);
+
+namespace ZP\LoyaltyProgram\Controller\Adminhtml\Program;
+
+use Magento\Framework\App\Action\HttpPostActionInterface;
+use Magento\Backend\App\Action\Context;
+use ZP\LoyaltyProgram\Api\LoyaltyProgramRepositoryInterface;
+use ZP\LoyaltyProgram\Model\LoyaltyProgram;
+use ZP\LoyaltyProgram\Model\LoyaltyProgramFactory;
+use Magento\Framework\Exception\LocalizedException;
+use ZP\LoyaltyProgram\Controller\Adminhtml\Program as AbstractProgramController;
+use Magento\Framework\Controller\ResultInterface;
+use ZP\LoyaltyProgram\Setup\Patch\Data\AddBasicPrograms as BasicProgramsConfig;
+
+class Save extends AbstractProgramController implements HttpPostActionInterface
+{
+    /**
+     * Save constructor.
+     * @param Context $context
+     * @param LoyaltyProgramFactory $programFactory
+     * @param LoyaltyProgramRepositoryInterface $programRepository
+     */
+    public function __construct(
+        Context $context,
+        private LoyaltyProgramFactory $programFactory,
+        private LoyaltyProgramRepositoryInterface $programRepository
+    ) {
+        parent::__construct($context);
+    }
+
+    /**
+     * Save action
+     *
+     * @return \Magento\Framework\Controller\ResultInterface
+     */
+    public function execute(): ResultInterface
+    {
+        /** @var \Magento\Backend\Model\View\Result\Redirect $resultRedirect */
+        $resultRedirect = $this->resultRedirectFactory->create();
+        $data = $this->getRequest()->getPostValue();
+
+        if ($data) {
+            $programId = $this->getRequest()->getParam(LoyaltyProgram::PROGRAM_ID);
+            if (empty($data[LoyaltyProgram::PROGRAM_ID])) {
```

Comment:
empty($data[LoyaltyProgram::PROGRAM_ID]  ? data[LoyaltyProgram::PROGRAM_ID] = null : $data[LoyaltyProgram::PROGRAM_ID] = (int)$programId

Reply:
$programId = $this->getRequest()->getParam(LoyaltyProgram::PROGRAM_ID);
  if (empty($data[LoyaltyProgram::PROGRAM_ID])) {
      $data[LoyaltyProgram::PROGRAM_ID] = null;
  } else {
      $programId = (int)$data[LoyaltyProgram::PROGRAM_ID];
      $data[LoyaltyProgram::PROGRAM_ID] = $programId;
  }
  
  I wiil not change this place because $programId variable is using further in different methods and i need to make it integer here once, and not to add (int) further in all places where it is using.

Reply:
use class property